fly/commands/pin_resource.go

```diff
@@ -0,0 +1,77 @@
+package commands
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/concourse/concourse/fly/commands/internal/displayhelpers"
+	"github.com/concourse/concourse/fly/commands/internal/flaghelpers"
+	"github.com/concourse/concourse/fly/rc"
+	"github.com/concourse/concourse/go-concourse/concourse"
+)
+
+type PinResourceCommand struct {
+	Resource flaghelpers.ResourceFlag `short:"r" long:"resource" required:"true" value-name:"PIPELINE/RESOURCE" description:"Name of the resource"`
+	Version  flaghelpers.JsonFlag     `short:"v" long:"version" description:"JSON string of at least one field of the version. The given key value pair(s) has to be an exact match but not all fields need to be provided. In the case of multiple resource versions matched, it will pin the latest one."`
```

Comment:
```suggestion
	Version  *atc.Version     `short:"v" long:"version" description:"JSON string of at least one field of the version. The given key value pair(s) has to be an exact match but not all fields need to be provided. In the case of multiple resource versions matched, it will pin the latest one."`
```